specs/phase1/shard-transition.md

```diff
@@ -64,7 +62,20 @@ def verify_shard_block_signature(beacon_state: BeaconState,
 
 ```python
 def shard_state_transition(shard_state: ShardState,
-                           block: ShardBlock) -> None:
+                           block: ShardBlock,
+                           validate_message: bool = True,
+                           beacon_parent_state: Optional[BeaconState] = None) -> bool:
+    if validate_message:
+        assert beacon_parent_state is not None
```

Comment:
Why make `beacon_parent_state` optional?

Reply:
Because we don't need `beacon_parent_state` if `validate_message` is False. i.e., only need to call `process_shard_block`.

In `get_shard_transition_fields` in v-guide, the `beacon_state` is the beacon head state, *not* the `beacon_parent_state` of the given shard blocks. The beacon parent state can only be obtained from the database store. I think it's better to make it clear that the caller doesn't have to pass the unused `beacon_parent_state` when `validate_message` is False.